### ISSUE TITLE
Thoughts on why it doesn't always solve

### ISSUE THREAD

**Issue by Dunkalunk** [IgniparousTempest/javascript-tri-peaks-solitaire-solver#3]
Thought I would open the discussion here instead of on the web page where people are frustrated and don't care about how hard it is to code something like this ha!

I was looking at the code and trying to have it solve a grandmaster that was giving me trouble, and I think I know why this solver can't always get through any deck.

1. Not every deck is solvable, that's a given, sometimes it just won't work - but the frustration to users comes from the fact that it's not returning anything saying it couldn't be solved.  I was seeing it when I had the code pulled locally, but the web page doesn't display when it fails that I could see.  So at the end if it returns the game is failed, display that end result so that they know and they don't think it's "crashing"

2. This is a complex but simple solver at the same time.  To me it looks like it's built very well (as far as reading through it, not saying I could write anything better, I mainly know Java, not JS).  But where I think the main problem lies is it lacks the ability to try NOT always selecting a sequential card.  I haven't been playing too long, but there have been plenty of decks where I realize I need a card later on.  Even if I could drop it from the peaks now, I know from multiple attempts that there's another card I need it for higher up.  This solver seems to always look for a sequential card and drops it no matter what.  So it would take something even more complex, like temporarily storing attempts as it loops, and having it try different options of leaving cards even if it could drop the card at a certain point.

Anyway, like I said, not saying I could write anything better, but maybe this would help someone who could write it further?  And maybe I'm totally wrong on number 2 haha.  I tried to put in some `console.log` lines to see if I could view where it was in looping through cards or see different results and couldn't figure out a good way to solve that grandmaster I was on.

Really cool solver either way!  I wish people would appreciate things people build on their own free time haha.

**Comment by Dunkalunk** [comment]
Thought I would add this is the board that I couldn't get solved:

7C TS 4D 3H 2C 4C 8S TC AH 2S 8C 9D QH 7S 6C 5H JD AS 5S 3D 5D 5C 4H JH 3S 8D 7D 7H 9H KD TD 9C JC 3C 9S 6H 6D 4S AC KS KH QS QC TH 6S AD 8H 2D KC QD JS 2H

**Comment by Dunkalunk** [comment]
Looks like I was wrong on number 2 I posted above.  I have used the solver multiple times where it could have just continued dropping cards but told me to draw new stock instead and ends up solving the deck for me.  So it's possible the deck I had in the comment was just unsolvable, or a really rare case where this solver couldn't get it.  Either way, still enjoying using it!  And I still think number 1 would be helpful.  Thanks!
